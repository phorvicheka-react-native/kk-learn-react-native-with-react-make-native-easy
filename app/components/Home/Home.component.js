import React, {Component} from 'react';
import {View, Text} from 'react-native';
import styles from './Home.component.style';
import TextArea from '../TextArea/TextArea.component';
import Counter from '../Counter/Counter.component';

class Home extends Component {
  render () {
    return (
      <View style={styles.container}>
        <Text> Please enter your note here</Text>
        <TextArea />
        <Counter initialCountValue={5} />
      </View>
    );
  }
}

export default Home;
