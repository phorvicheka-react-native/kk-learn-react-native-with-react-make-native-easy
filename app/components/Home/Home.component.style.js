import {StyleSheet} from 'react-native';
import theme from '../../styles/theme.style';

export default StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: theme.CONTAINER_PADDING,
    alignItems: 'center'
  },
  titleHeading: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    alignSelf: 'flex-start',
    padding: 10,
    fontWeight: theme.FONT_WEIGHT_BOLD,
  },
  titleTextInput: {
    padding: theme.TEXT_INPUT_PADDING,
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    alignSelf: 'stretch'
  },
  textAreaTitle: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    alignSelf: 'flex-start',
    padding: 10,
    fontWeight: theme.FONT_WEIGHT_LIGHT,
    fontStyle: 'italic'
  },
  textArea: {
    padding: theme.TEXT_INPUT_PADDING,
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    alignSelf: 'stretch',
    flex: 1
  }
});
