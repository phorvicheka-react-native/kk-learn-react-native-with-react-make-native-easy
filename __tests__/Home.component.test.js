import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../app/components/Home/Home.component';

describe('Home component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Home />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});