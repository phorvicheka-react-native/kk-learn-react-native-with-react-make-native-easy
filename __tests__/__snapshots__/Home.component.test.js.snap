// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text>
     Please enter your note here
  </Text>
  <TextInput
    allowFontScaling={true}
    multiline={true}
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 100,
          "width": 200,
        },
        undefined,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          +
        </Text>
      </View>
    </View>
    <Text>
      Count: 
      5
    </Text>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          -
        </Text>
      </View>
    </View>
  </View>
</View>
`;
